Assets/Scripts/TrainingArena.cs

```diff
@@ -87,6 +95,32 @@ private void InitializeArenaComponents()
 		Spawner_InteractiveButton.RewardSpawned += OnRewardSpawned;
 	}
 
+	/// <summary>
+	/// Provides a list of the arenas in the current config file that are preceeded by an arena with
+	/// the mergeNextArena property, so that we can avoid loading them when arenas are randomised.
+	/// </summary>
+	private List<int> GetMergedArenas()
+	{
+		List<int> mergedArenas = new List<int>();
+		int totalArenas = _environmentManager.getArenaCount();
+		ArenaConfiguration currentArena;
+		if (!_environmentManager.GetConfiguration(0, out currentArena))
+		{
+			Debug.LogError("Critical error: Failed to load arena configuration for arena 0");
+		}
```

Comment:
Aah yes good point. I'm not very keen on this pattern of call `_environmentManager.GetConfiguration`, check its output, and handle if it fails since we end adding the same handling for every call.

I think in previous commits they checked this and handled the case where the config wasn't found ([ref](https://github.com/Kinds-of-Intelligence-CFI/animal-ai-unity/blob/979e25d42201247f92394fcfd663d92060bffc75/Assets/Scripts/TrainingArena.cs#L117)). Doing it that way did have the benefit of handling the case where for whatever reason a config doesn't have a contiguous run of arena numbers starting from 0, but I think could probably mask errors too so I'm not too worried about having AAI throw in that case.

Since we're not doing that handling any more we probably don't need a 'this operation succeeded' bool at all. How about I push that handling down into the definition for `GetConfiguration` and let that function throw if it fails?

(As an aside when I'm using AAI I do find doing arena numbering a bit of a pain point. Perhaps we could a better way?)

Reply:
Arena numbering really is a pain and probably the culprit of a lot of the bugs I fixed earlier. However, this would need some research and would need dedicated time and effort into overhauling for a more robust, easier to handle arena handling logic.